### DISCUSSION TITLE
How to use Javascript in a practical way in Keycloak Quarkus?

### DISCUSSION THREAD

**Discussion by bcm0** [keycloak/keycloak#16667] [Q&A]
Do we really need a full keycloak restart after every change to the .jar archive?
If yes, where do you develop your Javascript?
Do you make a change, create the archive, put it in keycloak and restart it?
Do you know an easier way?

**Comment by bonnm** [comment]
I'm developing the Java Files under Windows, using Visual Studio or (for small changes) Notepadd++. Then I have a Powershell Script which packs all Files (*.js) in a zip (the meta-inf stuff is already in the zip), copies the zip to .jar and uploads the  jar to my Keycloak (Test) Server (Ubuntu Linux) using the standard SFTP client shipped with Windows. Then a (remotely by SSH executed) cp copies the jar to the Keycloak's deployments directory. As I am still on Keycloak 18.0.2 Wildfly, a restart of Keycloak is not necessary. When everything is fine, I do the same with my production servers as SFTP/SSH target.
